srml/babe/src/lib.rs

```diff
@@ -151,11 +155,15 @@ decl_module! {
 				.iter()
 				.filter_map(|s| s.as_pre_runtime())
 				.filter_map(|(id, mut data)| if id == BABE_ENGINE_ID {
-					<[u8; VRF_OUTPUT_LENGTH]>::decode(&mut data)
+					<(u64, [u8; VRF_OUTPUT_LENGTH])>::decode(&mut data)
 				} else {
 					None
 				}) {
-				Self::deposit_vrf_output(&i);
+				if i.0 != EpochIndex::get() {
+					// New epoch reported by validator
```

Comment:
can't it be derived from slot number? the `epoch` parameter seems redundant

Reply:
That would require getting the chain configuration into the runtime, though.  Not sure how to do that.

Reply:
`GenesisConfig`, no?
The node should be informed by the chain which is informed by the genesis config. At the moment, the node will draw information about configuration only at genesis. Later on we'll have it draw information at other points.